kratos/python_scripts/multistage_orchestrators/multistage_orchestrator.py

```diff
@@ -0,0 +1,182 @@
+import sys
+import importlib
+
+import KratosMultiphysics
+from KratosMultiphysics.project import Project
+from KratosMultiphysics.model_parameters_factory import KratosModelParametersFactory
+
+class MultistageOrchestrator():
+    """Base class for multistage orchestrators.
+    
+    This class is intended to serve as base for al the Kratos multistage orchestrators.
+    The purpose of orchestrators is to define the workflow of the multistage simulations.
+    This includes deciding when and how to save and load (aka restart) the simulation.
+    Run() method must be implemented in derived classes (see SequentialMultistageOrchestrator).
+
+    Member variables:
+    __project -- Current multistage simulation project container
+    """
+
+    def __init__(self, project : Project) -> None:
+        """Constructs a multistage orchestrator instance"""
+
+        # Store pointer to current project
+        # Note that the project already contains the multistage simulation settings
+        self.__project = project
+
+    def Run(self):
+        """Main function that runs the complete multistage simulation."""
+
+        err_msg = "Calling base MultistageOrchestrator Run() method. This must be implemented in derived orchestrators."
+        raise NotImplementedError (err_msg)
```

Comment:
```suggestion
    @abstractmethod
    def Run(self):
        """Main function that runs the complete multistage simulation."""
```

this is a more modern and safer solution (requires to derive the class from `ABC`)

Reply:
In my first implementation I did it like this. However, it gave me problems. If I'm nor remembering wrong, it required the entire class to be abstract, and we're implementing a bunch of methods in here.

Reply:
I agree with @philbucher , you can have bunch of method implemented, and bunch to be abstract. This is safer, so at interpretation time you get the error message rather than the run time.

This is not possible when you have a base class like `Kratos.Process` since the pybind abstract meta class and python abstract meta class is in conflict.

Reply:
OK. If so most probably I did something wrong. Let me give it another try.

Reply:
> This is not possible when you have a base class like Kratos.Process since the pybind abstract meta class and python abstract meta class is in conflict.

That's a very unfortunate limitation of the `abc` implementation in python. I opened an [issue](https://github.com/python/cpython/issues/102077) for it a while back, but it didn't get any attention :'(. I'd very much like to use `abc` with `pybind` classes, so I'm open to giving it another go or roll our own implementation if someone else is also interested.

Reply:
@philbucher @sunethwarna @matekelemen I did it again and it worked. I've discussed briefly with @roigcarlo to figure out what could have happened the last time we tried, we believe that is something related to the version before the refactor to orchestrators, long story short, nothing to be worried about.